### PR TITLE
Adding new cluster search setting docs

### PR DESCRIPTION
**Pull request by harshavamsi**
### Description
Adds docs around new cluster search setting

### Issues Resolved
Resolves https://github.com/opensearch-project/documentation-website/issues/8175

### Version
2.17

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _install-and-configure/configuring-opensearch/search-settings.md [hunk 1]
```diff
@@ -39,6 +39,8 @@ OpenSearch supports the following search settings:
 
 - `search.dynamic_pruning.cardinality_aggregation.max_allowed_cardinality` (Dynamic, integer): Determines the threshold for applying dynamic pruning in cardinality aggregation. If a field’s cardinality exceeds this threshold, the aggregation reverts to the default method. This is an experimental feature and may change or be removed in future versions.
 
+- `search.keyword_index_or_doc_values_enabled` (Dynamic, Boolean): Determined whether to use `IndexOrDocValuesQuery` when running `MultiTermQueries` on `Keyword` fields. Default value is `false`.
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/search-settings.md:42]:
```suggestion
- `search.keyword_index_or_doc_values_enabled` (Dynamic, Boolean): Determines whether to use `IndexOrDocValuesQuery` when running `multi_term` queries on `keyword` fields. Default value is `false`.
```

**Review by kolchfa-aws — COMMENTED** [review]
Thank you, @harshavamsi! We want to avoid Java class names in documentation and just explain what the term means. Could you please clarify what is meant by "index or doc_values query"?

**Comment by harshavamsi** [comment]
@kolchfa-aws `IndexOrDocValuesQuery` essentially will make a decision during query time to decide whether to use the index structure to run the query or the doc_values. It comes from Apache Lucene, I wasn't sure how we could better describe here since it really is an implementation detail and the class name is as verbose as it gets.

**Comment by kolchfa-aws** [comment]
Thank you! I updated my suggestion.

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @harshavamsi! LGTM

**Review by natebower — APPROVED** [review]
LGTM

**Pull request merged**